sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberFormatter.scala

```diff
@@ -149,6 +156,21 @@ class NumberFormatter(originNumberFormat: String, isParse: Boolean = true) exten
     } else if (invalidSignPosition(DOLLAR_SIGN)) {
       TypeCheckResult.TypeCheckFailure(
         nonFistOrLastCharInNumberFormatError(s"'$DOLLAR_SIGN'"))
+    } else if (!isParse && normalizedNumberFormat.exists(_ == COMMA_SIGN)) {
```

Comment:
As of this PR [1], the `NumberFormatter` class is deleted now :) It looks like that got merged after this PR started but before now. How do we want to handle this? Should we (a) merge latest commits from master including [1] and then update this PR to use the `ToNumberParser` class instead, or (b) keep this `NumberFormatter` class around just for the `to_char` function and delete all the code related to parsing strings to `Decimal` values?

Either approach sounds OK, WDYT?

[1] https://github.com/apache/spark/pull/36066

Reply:
@cloud-fan What about your opinion ?

Reply:
Can we re-implement `to_char` with the new parser we added for `to_number`? These 2 functions must be consistent.

Reply:
OK

Reply:
@beliefer Sure I can be happy to help. I can prepare the change and then send it to yourself and Wenchen for review. Sorry for the churn in the code making things complicated.